src/AzureClient/AzureClient.cs

```diff
@@ -129,6 +129,7 @@ public class AzureClient : IAzureClient
                 baseEngine.RegisterDisplayEncoder(new AzureClientErrorToTextEncoder());
                 baseEngine.RegisterDisplayEncoder(new DeviceCodeResultToHtmlEncoder());
                 baseEngine.RegisterDisplayEncoder(new DeviceCodeResultToTextEncoder());
+                baseEngine.RegisterDisplayEncoder(new ResourceEstimationToHtmlEncoder(logger));
             }
```

Comment:
I'd suggest using ActivatorUtils here to inject dependencies, rather than manually creating a logger. That will make sure the logger has the right metadata as to which service is generating log entries.